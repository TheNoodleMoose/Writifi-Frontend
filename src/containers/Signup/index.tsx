import React from "react";

const Signup = () => {
  return (
    <div>
      <p>This is the signup page</p>
    </div>
  );
};

export default Signup;
