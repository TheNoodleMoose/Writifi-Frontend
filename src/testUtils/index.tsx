import React, { ReactElement, ReactNode } from "react";

import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export const renderWithContext = (component: ReactElement) => {
  const history = createMemoryHistory();
  const wrapper = ({ children }: { children?: ReactNode }) => (
    <Router history={history}>{children}</Router>
  );

  const renderResults = render(component, { wrapper });
  return {
    ...renderResults,
    snapshot: () => {
      expect(renderResults.container).toMatchSnapshot();
    }
  };
};
