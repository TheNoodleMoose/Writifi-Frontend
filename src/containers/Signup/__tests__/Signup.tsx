import React from "react";
import { renderWithContext } from "../../../testUtils";

import Signup from "..";

it("renders correctly", () => {
  renderWithContext(<Signup />).snapshot();
});
