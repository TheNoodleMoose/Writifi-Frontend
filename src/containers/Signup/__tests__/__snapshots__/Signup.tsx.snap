// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <p>
      This is the signup page
    </p>
  </div>
</div>
`;
